### ISSUE TITLE
Add support for lax mode

### ISSUE THREAD

**Issue by pdesaulniers** [teal-language/vscode-teal#8]
Right now, the extension only checks .tl files. It could also check .lua files in 'lax' mode.

**Comment by pdesaulniers** [comment]
I ran `tl check` on some Lua files, and I received lots of errors which weren't very useful.

Once https://github.com/teal-language/tl/issues/89 is implemented, we might be able to use `tl check` as a `luacheck` replacement.

**Comment by tommitytom** [comment]
Just a note, that this is excellent for general lua programming: https://marketplace.visualstudio.com/items?itemName=sumneko.lua

**Comment by pdesaulniers** [comment]
You're right, it's very good! I was still using https://github.com/trixnz/vscode-lua, which is much less feature-rich...

**Comment by tommitytom** [comment]
A further note on the previously mentioned extension, this is an interesting development: 

https://github.com/sumneko/lua-language-server/wiki/Annotations
https://github.com/sumneko/lua-language-server/wiki/Plugins

**Comment by dipyalov** [comment]
It seems like tree-sitter-teal grammar doesn't support lua files and generates wrong tokens, so intellisense won't work correctly
